src/components/filter/Filter.tsx

```diff
@@ -0,0 +1,38 @@
+import { useContext, SetStateAction } from "react";
+import * as Styled from "./Filter.styled";
+import { TeamEntriesContext } from "../../context/TeamEntriesProvider";
+
+interface FilterProps {
+  setCurrentClient: React.Dispatch<SetStateAction<string>>;
+}
+
+export const Filter = ({ setCurrentClient }: FilterProps) => {
+  const { teamEntries, setTeamEntries } = useContext(TeamEntriesContext);
+
+  const handleFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setCurrentClient(event.target.value);
+  };
+
+  const onlyClients = teamEntries.map(function (entry) {
+    return entry.client;
+  });
+  const formattedClients = [...new Set(onlyClients)];
```

Comment:
You could even combine your two variables to just the one: 
```suggestion
  const clients = [...new Set(teamEntries.map((entry) => entry.client))];
```
But it's also fine to keep these two functions separate if you feel like it makes your code more readable.